Source/Scene/CreditDisplay.js

```diff
@@ -77,113 +76,275 @@ define([
+            lightboxCredits.style.marginTop = Math.floor((height - lightboxCredits.clientHeight) * 0.5) + 'px';
+            that._lastViewportHeight = height;
+        }
     }
 
     /**
      * The credit display is responsible for displaying credits on screen.
      *
      * @param {HTMLElement} container The HTML element where credits will be displayed
      * @param {String} [delimiter= ' • '] The string to separate text credits
+     * @param {HTMLElement} [viewport=document.body] The HTML element that will contain the credits popup
      *
      * @alias CreditDisplay
      * @constructor
      *
      * @example
      * var creditDisplay = new Cesium.CreditDisplay(creditContainer);
      */
-    function CreditDisplay(container, delimiter) {
+    function CreditDisplay(container, delimiter, viewport) {
         //>>includeStart('debug', pragmas.debug);
-        if (!defined(container)) {
-            throw new DeveloperError('credit container is required');
-        }
+        Check.defined('container', container);
         //>>includeEnd('debug');
+        var that = this;
+
+        viewport = defaultValue(viewport, document.body);
+
+        var lightbox = document.createElement('div');
+        lightbox.className = 'cesium-credit-lightbox-overlay';
+        lightbox.style.display = 'none';
+        lightbox.style.zIndex = '5';
+        lightbox.style.position = 'absolute';
+        lightbox.style.top = 0;
+        lightbox.style.left = 0;
+        lightbox.style.width = '100%';
+        lightbox.style.height = '100%';
+        lightbox.style.backgroundColor = 'rgba(80, 80, 80, 0.8)';
+        viewport.appendChild(lightbox);
+
+        var lightboxCredits = document.createElement('div');
+        lightboxCredits.style.backgroundColor = '#303336';
+        lightboxCredits.style.color = '#edffff';
+        lightboxCredits.style.position = 'relative';
+        lightboxCredits.style.minHeight = lightboxHeight + 'px';
+        lightboxCredits.style.margin = 'auto';
+        lightboxCredits.className = 'cesium-credit-lightbox';
+        lightbox.appendChild(lightboxCredits);
+        lightbox.onclick = function(event) {
+            if (event.target === lightboxCredits) {
+                return;
+            }
+            that.hideLightbox();
+        };
+
+        var title = document.createElement('div');
+        title.textContent = 'Data provided by:';
+        title.style.padding = '20px 20px 0 20px';
+        lightboxCredits.appendChild(title);
+
+        var closeButton = document.createElement('a');
+        closeButton.onclick = this.hideLightbox.bind(this);
+        closeButton.innerHTML = '&times;';
+        closeButton.style.fontSize = '18pt';
+        closeButton.style.cursor = 'pointer';
+        closeButton.style.position = 'absolute';
+        closeButton.style.top = '0';
+        closeButton.style.right = '6px';
+        closeButton.onmouseover = function() {
```

Comment:
Would be great to move all this styling to stylesheets.  This could be a `:hover` style instead of a mouse-specific JS event.

Reply:
It would be great.  We can't do that though because we can't require that people include our CSS

Reply:
The other way to do this could be to use JS to inject a stylesheet in the head element, if it's not already present, above other stylesheets for easy overriding.  That's a bit of effort but is known to work well.

Reply:
Not required I guess, since apps can use `!important` on anything here that does have a className.

Reply:
Oh good idea, I didn't think about that.  Yeah, overriding these would be a pain if someone needed to.

Reply:
> It would be great. We can't do that though because we can't require that people include our CSS

I'm confused by what this statement means.  They have to include our css in order to use any of our widgets.

Reply:
> They have to include our css in order to use any of our widgets.

Hmm, yeah, `CesiumWidget` needs its stylesheet for its error dialog and such, even without the viewer.  Maybe these styles should go there after all.

Reply:
I see `CesiumWidget.css` has an error panel with `z-index: 99999;` to which `git` says:

> Blame Ed Mackey (51 months ago)

Great.  This was apparently needed for exactly the same reason as the new z-index, to overcome the viewer's outer widgets which appear after CesiumWidget in the DOM tree.

Reply:
![image](https://user-images.githubusercontent.com/1495196/33355296-e724728a-d484-11e7-8f99-539284b3aeaf.png)